File: src/main/java/burrow/furniture/hoard/command/SetCommand.java
package burrow.furniture.hoard.command;

import burrow.core.command.Command;
import burrow.core.command.CommandContext;
import burrow.core.command.CommandType;
import burrow.furniture.hoard.HoardFurniture;
import org.jetbrains.annotations.NotNull;
import picocli.CommandLine;

import java.util.Map;

@CommandLine.Command(name = "set", description = "Set a property for a specific entry.")
@CommandType(HoardFurniture.COMMAND_TYPE)
public class SetCommand extends Command {
    @CommandLine.Parameters(index = "0", description = "The ID of the entry to update.")
    private Integer id;

    @CommandLine.Parameters(index = "1", description = "The key to set.")
    private String key;

    @CommandLine.Parameters(index = "2", description = "The value to set.")
    private String value;

    public SetCommand(@NotNull final CommandContext commandContext) {
        super(commandContext);
    }

    @Override
    public Integer call() {
        final var hoardFurniture = use(HoardFurniture.class);
        final var entry = hoardFurniture.getHoard().setProperties(id, Map.of(key, value));
        buffer.append(hoardFurniture.entryToString(entry, commandContext));

        return CommandLine.ExitCode.OK;
    }
}

File: src/main/kotlin/carton/hoard/command/SetCommand.kt
package burrow.carton.hoard.command

import burrow.carton.hoard.Hoard
import burrow.carton.hoard.checkId
import burrow.carton.hoard.checkPairs
import burrow.kernel.terminal.*

@BurrowCommand(
    name = "set",
    header = ["Sets properties for a specific hoard entry by ID."]
)
class SetCommand(data: CommandData) : Command(data) {
    @Parameters(
        index = "0",
        description = [
            "The unique ID of the entry to modify. Must be a positive integer."
        ]
    )
    var id = 0

    @Parameters(
        index = "1..*",
        description = [
            "The keys and values to set as properties for the specified entry."
        ]
    )
    private var pairs: Array<String> = emptyArray()

    override fun call(): Int {
        if (!checkId(id, stderr)) {
            return ExitCode.USAGE
        }

        if (!checkPairs(pairs, stderr)) {
            return ExitCode.USAGE
        }

        val properties = mutableMapOf<String, String>()
        for (i in pairs.indices step 2) {
            properties[pairs[i]] = pairs[i + 1]
        }

        use(Hoard::class).storage.operate(id) { set(properties) }

        return dispatch(EntryCommand::class, listOf(id))
    }
}